app/src/composables/useSiwe.ts

```diff
@@ -25,37 +23,53 @@ function createSiweMessageCreator(address: string, statement: string, nonce: str
   })
 }
 
-async function siwe() {
-  const { addErrorToast } = useToastStore()
-
-  try {
-    isProcessing.value = true
-    const address = await ethersJsAdapter.getAddress()
-    const response = await useCustomFetch<string>(`user/nonce/${address}`).get().json()
-    const nonce = response.data.value.nonce
-    const statement = 'Sign in with Ethereum to the app.'
-    const siweMessageCreator = createSiweMessageCreator(address, statement, nonce)
-    const siweAuthService = new SIWEAuthService(siweMessageCreator, ethersJsAdapter, siweAuthApi)
-
-    await siweAuthService.authenticateUser()
-    const result = await useCustomFetch<string>(`user/${address}`).get().json()
-    const userData: Partial<User> = result.data.value
-    useUserDataStore().setUserData(
-      userData.name || '',
-      userData.address || '',
-      userData.nonce || ''
-    )
-    useUserDataStore().setAuthStatus(true)
-
-    router.push('/teams')
-    isProcessing.value = false
-  } catch (error: any) {
-    isProcessing.value = false
-    log.error(parseError(error))
-    addErrorToast(parseError(error))
+export function useSiwe() {
+  const isProcessing = ref(false)
+  const error: Ref<string | null> = ref(null)
+
+  async function siwe() {
+    try {
+      isProcessing.value = true
+      const address = await ethersJsAdapter.getAddress()
+      const {
+        //isFetching: isFetchingNonce,
+        error: fetchError,
+        data: nonce
+        //execute: executeFetchNonce
+      } = await useCustomFetch<string>(`user/nonce/${address}`).get().json()
+
+      if (fetchError.value) throw new Error(errorMessage.value)
+
```

Comment:
You can handle the error here

by formatting the error here.
So if you have fetch Error, Just create the toast. with the formatted message.

Reply:
Isn't easier to just handle the error once (in the catch block) instead of every time you call code that can throw an error. In this you case just format the message and throw it to be handled in the catch block

Reply:
Ooh okay, I get it now. What if we user the logger for that, like this. In that case if an error occurs you have a stack trace that shows:

```
export const useCustomFetch = createFetch({
  baseUrl: `${BACKEND_URL}/api/`,
  combination: 'chain',
  options: {
    async beforeFetch({ options }) {
      const token = AuthService.getToken()
      options = {
        mode: 'cors',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`
        }
      }
      return { options }
    },
    onFetchError({ data, response, error }) {
      const status = response?.status ?? 0
      responseSatus.value = status
      console.log('[useCustomFetch] error: ', error)
      //Log the error
      log.error(
        `${
          status
            ? `Error ${status}: ${getFetchErrorMessage(status)}`
            : `Network Error: ${error?.message ?? 'An unknown error occurred'}`
        }`
      )
      return { data, response, error }
    }
  }
})
```

Reply:
When we have an error it appear in the console, in the case of fetch error.
The request have two purpose. 

- Readability for the dev
- And user experience.

And about your question to handle the error once. There is another way to do it. And make the code more efficient.


You can simply put the useCustom fetch line outside the function, and call the execute inside. 
So you can create a compute state that will return a build error. And add a watch inside that show the toast on error.

Reply:
Okay, I don't quite understand the implementation that returns a build error also, `ethersJSAdapter` and `siweAuthService` can also throw errors. Do you want to have a try catch block around each of these?